cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -259,13 +311,27 @@ visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodeList() {
       const ::geometry_msgs::Point node_point =
           ToGeometryMsgPoint(node_id_data.data.global_pose.translation());
       marker.points.push_back(node_point);
+      bool is_switch_node =
```

Comment:
I think without this bool it will be more readable.
Just update `double marker_alpha` every time.
Do PushAndResetLineMarker if that is different from `marker.color.a`.